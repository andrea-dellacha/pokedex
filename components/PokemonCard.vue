<template>
  <div class="cardPokemon" @click="$router.push(`/${pokemon.id}`)">
    <div class="card__image">
      <img
        :id="`pokemon#${pokemon.id}`"
        :src="`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemon.id}.svg`"
        :alt="pokemon.name"
      />
    </div>
    <div class="card__title">
      {{ pokemon.name | capitalize }}
    </div>
    <div class="card__id">
      {{ pokemon.id | formatPokemonId }}
    </div>
  </div>
</template>

<script>
export default {
  name: 'PokemonCard',
  props: {
    pokemon: {
      type: Object,
      required: true,
    },
  }
}
</script>

<style scoped>
.cardPokemon {
  height: 300px;
  width: 200px;
  background-color: white;
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
  border-radius: 8px;
  display: grid;
  grid-template-rows: 200px 40px 40px;
  padding: 10px;
}

.card__image, .card__id, .card__title {
  display: flex;
  justify-content: center;
  align-items: center;
}

.card__image img {
  max-width: 180px;
  max-height: 180px;
}

.card__title {
  font-size: 24px;
}

/* Media queries*/
@media screen and (max-width: 600px) {
  .cardPokemon {
    height: 150px;
    width: 100px;
    grid-template-rows: 100px 20px 20px;
    font-size: var(--font-size);
  }

  .card__image img{
    max-width: 90px;
    max-height: 90px;
  }

  .card__title {
    font-size: 16px;
  }
}
</style>
