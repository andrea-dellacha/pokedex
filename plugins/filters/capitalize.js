import Vue from 'vue'

export default function capitalize (value) {
  if (!value) { return '' }
  value = value.toString()

  return value.charAt(0).toUpperCase() + value.slice(1)
}

Vue.filter('capitalize', capitalize)
