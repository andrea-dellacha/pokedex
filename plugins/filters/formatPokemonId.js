import Vue from 'vue'

export default function formatPokemonId (value) {
  if (value) {
    switch (value.toString().length) {
      case 1:
        return `00${value}`
      case 2:
        return `0${value}`
      case 3:
        return `${value}`
    }
  }  
}

Vue.filter('formatPokemonId', formatPokemonId)
