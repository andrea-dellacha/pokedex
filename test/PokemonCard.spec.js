import { shallowMount, createLocalVue } from '@vue/test-utils'
import PokemonCard from '@/components/PokemonCard.vue'
import capitalize from '@/plugins/filters/capitalize'
import formatPokemonId from '@/plugins/filters/formatPokemonId'

describe('PokemonCard', () => {
  const localVue = createLocalVue()

  // Filter import
  localVue.filter('capitalize', capitalize)
  localVue.filter('formatPokemonId', formatPokemonId)

  const wrapper = shallowMount(PokemonCard, {
    propsData: {
      pokemon: {
        id: 5,
        name: 'charmander'
      }
    }
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })
})
