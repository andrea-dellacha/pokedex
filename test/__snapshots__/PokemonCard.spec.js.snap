// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard renders correctly 1`] = `
<div
  class="cardPokemon"
>
  <div
    class="card__image"
  >
    <img
      alt="charmander"
      id="pokemon#5"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/5.svg"
    />
  </div>
   
  <div
    class="card__title"
  >
    
    Charmander
  
  </div>
   
  <div
    class="card__id"
  >
    
    005
  
  </div>
</div>
`;
